src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_help.py

```diff
@@ -274,6 +274,18 @@
     type: command
     short-summary: Unregister a provider.
 """
+helps['provider operation'] = """
+    type: group
+    short-summary: get provider operations.
+"""
+helps['provider operation show'] = """
+    type: command
+    short-summary: Get an individual provider's operations.
+"""
+helps['provider operation list'] = """
+    type: command
+    short-summary: Get operations from all providers.
```

Comment:
You can simply do:
```Python
if provider:
  return [client.get(...)]
else:
  return client.list(...)
```
In this case both "show" and list return lists. Show doesn't return an individual object like VM show does. It returns a list of things for single provider.  That's the same paradigm as list/list-all.

At least, that's what I gather from the help text.  What does the output look like for "show" vs. "list"?